provision-cluster/lib/kubeception.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+const core = require('@actions/core');
+const httpClient = require('@actions/http-client');
+const httpClientLib = require('@actions/http-client/lib/auth.js');
+
+function getHttpClient() {
+	const userAgent = 'datawire/provision-cluster'
+
+  const kubeceptionToken = core.getInput('kubeceptionToken');
+  if (!kubeceptionToken) {
+    throw Error(`kubeceptionToken is missing. Make sure that input parameter kubeceptionToken was provided`);
+  }
+
+	const credentialHandler = new httpClientLib.BearerCredentialHandler(kubeceptionToken);
+  return new httpClient.HttpClient(userAgent, [credentialHandler]);
+}
+
+async function createKluster(name, version) {
+  if (!name) {
+    throw new Error('Function createKluster() needs a Kluster name');
+  }
+
+  if (!version) {
+    throw new Error('Function createKluster() needs a Kluster version');
+  }
+
+  const kubeceptionToken = core.getInput('kubeceptionToken');
+  if (!kubeceptionToken) {
+    throw Error(`kubeceptionToken is missing. Make sure that input parameter kubeceptionToken was provided`);
+  }
+
+  const client = getHttpClient();
+
+	const oneDay = 86400
+	const response = await client.put(`https://sw.bakerstreet.io/kubeception/api/klusters/${name}?version=${version}&wait=true&timeoutSecs=${oneDay}`);
```

Comment:
I know this version isn't intended to have retry/error handling yet, but in case you are tempted to use `wait=true` *and* retry at the same time, I'll mention that most (all?) of the CI clients have ended up moving away from that strategy and instead using a simple polling loop (e.g. 1 request per second) without the wait parameter.

The reason for that movement is kinda interesting, since you might think the optimal strategy would be to use `wait=true` parameter *and* retry. That's certainly what I initially attempted with the first CI client, however it turns out that if you use `wait=true` then your retry code path only gets executed when the tcp connection fails, which is *much* less frequently than once per second. That makes it a lot easier to break that retry path. If instead you sleep on the client side and don't use `wait=true` you end up with a retry code path that will be exercised almost every time the code is used, and so it's much harder to break that code path.